### ISSUE TITLE
No $rasp_version 11 check

### ISSUE THREAD

**Issue by aiuhjv1** [mudpi/installer#11]
It looks like your install script hasn't been updated for Raspbian version 11.  As such $php_version isn't filled which is causing some failures in the install process.

**Comment by QuartzIMG** [comment]
I may be running into this as well, though please let me know if this is an inappropriate location for this:

Im on 
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian

fresh install and freshly updated/rebooted, gets to this point and fails:

```
zip is already the newest version (3.0-12).
zlib1g-dev is already the newest version (1:1.2.11.dfsg-2+deb11u1).
libc6-dev is already the newest version (2.31-13+rpt2+rpi1+deb11u2).
python3-pip is already the newest version (20.3.4-4+rpt1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Main Depepencies Successfully Installed
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
ffmpeg is already the newest version (7:4.3.3-0+rpt3+deb11u1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: RPi.GPIO in /usr/lib/python3/dist-packages (0.7.0)
sudo: php: command not found
MudPi Install Error: Problem installing composer

```

**Comment by Awesomeloper** [comment]
I ran into the same issue with Bullseye. I tried pushing through the issue by manually installing php and php-cli, but ran into more issues down the road with "MudPi Install Error: Unable to create redis.ini file in php/mods-available".

**Comment by Awesomeloper** [comment]
Have you already tested it and gotten it working? I cloned your forked version and ran the installer but it wasn't able to install dependencies.

**Comment by ajisrael** [comment]
Yeah I was able to get it working, but I don't recall if there was anything that I did before running the installer. I might have installed redis or some other dependencies manually prior to the install script. Looking at the error you have you can see that the path ```php/mods-available``` where it's trying to create the redis.ini file is an invalid path. There should be a version after php (```php/8.1/mods-available```) which my fork will solve. I'd be happy to try and help solve the dependencies issue if you want to share some of the error's you're seeing and what dependencies are failing to install.

**Comment by ajisrael** [comment]
@QuartzIMG I think you'll just have to install php separately. Then run the install script in my forked repo and you should be good to go.

**Comment by Awesomeloper** [comment]
PHP dependencies don't install. Says unable to locate php8.1. I'm trying to install it on an RPi zero w (not second gen). I don't believe 8.1 is supported for ARMv6. Should I attempt to install the supported php dependencies before running the installer?

**Comment by Awesomeloper** [comment]
Since it seems the installer forces PHP V8, I tried on a newer RPi and found the assistant (specifically compose.json) is requiring PHP V7. I would've posted this on that product, but issues are turned off. This whole suite needs addressed for bullseye. Support for ArmV6 seems to have been unintentionally removed by EOL PHP versions

**Comment by ajisrael** [comment]
Yeah I would just install php for your system separately, identify the version number and update that value in the install script. That should help resolve all the paths you need for the other dependencies/programs installed by the script.

**Comment by JimenezGSide** [comment]
Hi
I cant get this to install. I ran the installer, seemed ok but when I run Mudpi i get 'Config file doesnt exist at {config_path}? Any ideas?

**Comment by Awesomeloper** [comment]
Did you make the config file? You have to do it manually

**Comment by JimenezGSide** [comment]
Thanks for your help and please forgive my noob-ness.....but how do i do that?

**Comment by Awesomeloper** [comment]
No worries! Check out https://mudpi.app/docs/installation it has the steps. I believe there is also a config creator built into the program, or on the MudPi website, there is a configuration tool. Google search "MudPi config tool" and it should pop up for you.

**Comment by JimenezGSide** [comment]
This isn't easy to setup for your average gardener!
I have managed to get it installed by first installing redis using this guide: https://amalgjose.com/2020/08/11/how-to-install-redis-in-raspberry-pi/
Then I installed php using this guide: https://pimylifeup.com/raspberry-pi-latest-php/
Then I ran the installer by ajisrael found above.
Sorted the config file once it was installed by watching this: https://www.youtube.com/watch?v=rAGryBYLJIg
Now when I try to go to the UI via the IP address in a web browser i get "Welcome to nginx!....further config is required...". Pulling my hair out (...or would if i had any).
